src/services/schemas/SchemasService.ts

```diff
@@ -62,7 +62,11 @@ export class SchemaService {
           });
         },
         error => deferred.reject(error))
-      .then(() => deferred.resolve(), error => deferred.reject(error));
+      .then(() => {
+        this.getProtoSchemaErrors(container, schema.fileName).then((error) => {
```

Comment:
You no longer resolve the deferred promise, so this doesn't work in the normal case.